frontend/src/api/api.ts

```diff
@@ -460,6 +460,8 @@ export const Metrics = {
 export const Events = {
   getEventPreviews: (c: string): Promise<EventPages> =>
     requests.get("app/events/", { params: { c } }),
+  searchEventNames: (distinct: boolean): Promise<EventPages> =>
```

Comment:
Ideally, we have two additional methods:

one for looking up event name : list of property name map, and another for feeding event name + filters in to get a list of matching events. A bit confused as to what this method does exactly, looks like it just gets the top 10 events?

Reply:
Judging from it, it feels like its doing a db search and returns top ten as uou pointed.

also as a side consideration, think we should extend this with a fine grained search library like elasticsearch? Could be far more efficient but havent thought too deeply into this

Reply:
elasticsearch definitely out of scope for this, want to keep it simple atm. 

The two methods we need related to this are 
1. event to property name mapping  (the snippet in https://github.com/uselotus/lotus/pull/622#pullrequestreview-1323124772 should solve this using pg functions at the db level)
2. filter the events and return 10 of them for preview